dcp/api/job.py

```diff
@@ -1,13 +1,117 @@
 import pythonmonkey as pm
+import cloudpickle
+import dill
+import asyncio
+from ..js import utils
+from .. import dry
+from .job_serializers import (
+    default_serializers,
+    serialize,
+    deserialize,
+    convert_serializers_to_arguments,
+    validate_serializers
+)
+from .job_env import convert_env_to_arguments
+from .job_modules import convert_modules_to_requires
+from .job_fs import JobFS
+from collections.abc import Iterator
+from types import FunctionType
+import urllib
+from .pyodide_work_function import get_work_function_string
 
 def job_maker(super_class):
     class Job(super_class):
-        #def exec(self, *args, **kwargs):
-        #    # TODO change behaviour to match spec
-        #    #print("overidden by Job hook")
+        def __init__(self, job_js):
+            super().__init__(job_js)
+            self.js_ref.worktime = 'pyodide'
 
-        def wait(self, *args, **kwargs):
-            pass
+            self._wrapper_set_attribute("serializers", default_serializers)
+            self._wrapper_set_attribute("env", {})
+            self._wrapper_set_attribute("modules", [])
+            self._wrapper_set_attribute("fs", JobFS())
+            self._wrapper_set_attribute("_exec_called", False)
+            self.aio.exec = self._exec;
+            self.aio.wait = self._wait;
+
+        def _before_exec(self, *args, **kwargs):
+            if not self.js_ref.worktime == "pyodide":
+                pass
```

Comment:
this is a no-op `pass` just means "do nothing" https://www.geeksforgeeks.org/python-pass-statement/ --- so all the code below is still executed

we can add support for alternative worktimes later, doesn't have to be in this MR

Reply:
https://github.com/Distributive-Network/bifrost2/issues/25

Added issue here